pkg/virt-launcher/virtwrap/api/schema_test.go

```diff
@@ -182,6 +182,31 @@ var exampleXMLppc64le = `<domain type="kvm" xmlns:qemu="http://libvirt.org/schem
   <iothreads>2</iothreads>
 </domain>`
 
+var _ = Describe("XML marshal of domain device", func() {
+	It("should not add user alias prefix to the name of a non-user-defined alias", func() {
+		alias := &Alias{Name: "alias0", userDefined: false}
```

Comment:
> Thank you for the contribution!
> 
> I am mainly missing the reasoning to make this change.
> For example, what is the limitation of the current implementation, why we need both "regular" and user-defined alias versions, etc.
> 
> This information needs also to land on the commit message so future us will be able to easily track from where the change came from (and why).
> 
> This also seems to be a big change, covering many source files. I would suggest to split it in smaller commits so it can be easier to review.
> For example, proposing the new API and still keeping backward comparability would be nice to gradually move the change forward and dropping the "old" API only at the last commit. Alternatively, just splitting it even if each commit is not self-contained will be useful, as we can focus on each change individually (especially the one who changes the API).

Hi @EdDev

The reason for this PR was that we have seen that some devices' alias names got evolved into things like `ua-let0` (from `tablet0`), `ua-tio` (from `virtio`) in downstream tests involving Windows VM migration and shutdown. So the speculated lifecycle of these alias names, `tablet0` as an example, would likely be: 
      ```tablet0 --UnmarshalXML--> let0 --MarshalXML--> ua-let0``` 

For the said bug scenario, we could not quite figure out why the "unmarshalled" alias name, e.g. `tablet0`, was originated. But this fix will prevent such short "unmarshalled" alias name got trimmed shorter, leading to multiple devices having the same alias name. 

On why we use the XML marshal/unmarshal overloading to add the 'ua-', I do not know the exact original design reason for choosing such layered design pattern.

Hope this addresses aspects of your comments.

Reply:
The references to string constants are cleaned up.